### ISSUE TITLE
Ignoring version conflict for "operation" INDEX on document

### ISSUE THREAD

**Issue by izzie88** [confluentinc/kafka-connect-elasticsearch#500]
Hi,

I keep getting the error **"WARN   ||  Ignoring version conflict for operation INDEX on document version -1 in index"** on one of my topics.
I use the same settings with other topics on kafka connect but i dont get the error.

I don't know what the issue might be.

Please help

**Comment by tporeba** [comment]
I see the same on my instance. This issue seems to explain the problem, but unfortunately no solution is provided: https://github.com/confluentinc/kafka-connect-elasticsearch/issues/485

**Comment by ramyogi7283** [comment]
we do not have solution how to solve this or any suggestion how to avoid this issue ?

**Comment by oneryalcin** [comment]
Yes, expected behaviour must be to replace the document with tghe same id rather than throwing error. This is I think critical

**Comment by ramyogi7283** [comment]
write.method=UPSERT works find for not sending version external. But as per the code. DELETE still sends version so it is causing the DELETE ignoring issue. when record value NULL. I am confused this logic. How to avoid DELETE with EXTERNAL version send and if change UPSERT then again index or updating documents causing issue.

    final String id = config.shouldIgnoreKey(record.topic())
        ? String.format("%s+%d+%d", record.topic(), record.kafkaPartition(), record.kafkaOffset())
        : convertKey(record.keySchema(), record.key());

    // delete
    if (record.value() == null) {
      return maybeAddExternalVersioning(new DeleteRequest(index).id(id), record);
    }
